packages/evm/contracts/Yaru.sol

```diff
@@ -1,70 +1,70 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 pragma solidity ^0.8.17;
 
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
```

Comment:
`Yaru` was not intended to be `Ownable`.

Similar to `Hashi`, I'd prefer that an ownable version of this be a contract that either inherits or reads from this contract.

Two possible solutions that don't require the contract to be ownable:

1. Have a separate `Yaru` deployment for each network.
2. Assume `Yaho` is always deployed to the same address on all networks and leave it to the adapters to specify the network ID that the message came from.

Reply:
I agree with you in principle @auryn-macmillan, however note that once initialized it cannot get changed (because of the initializeForChainId method).
Given that, option 1. would imply the same trust model (you verify before deployment what the address is, and that it is initialized correctly) while duplicating code and overcomplicating the development process for Hashi users, while 2. doesn't seem to be something guaranteed to work on all networks (i.e. zksync?).

I still feel like this approach is a better balance. Ownable is just used to avoid anyone to mess up with yet uninitialized network references (as new networks could be added later), but doesn't really require trust as long as you verify before deploying your code that the network you want to use has indeed been added to Hashi and safely initiatlized. Once set, it cannot be changed by anyone, similarly to any initializable code.

Reply:
Right, but using `Ownable`, however small, now enshrines some entity as THE thing that can make canonical changes to the base layer, which is explicitly counter to the original goal of hashi.

Another way to solve for this might be to offload this to the `IOracleAdapter`. Essentially, each oracle adapter would define what it views as the cannonical `Yaho` deployment for any given network.